### PR TITLE
musl: add support for search.h

### PR DESCRIPTION
**Pull request by msabwat**
This is an attempt to add support for codebases that want to use functions in search.h.
I added a very simple test that I found in the man page, I can add more tests if needed.

### CODE REVIEW

File: tests/test_other.py [hunk 1]
```diff
@@ -416,6 +416,10 @@ def test_multiple_sources(self):
     self.assertNotExists(test_file('twopart_main.o'))
     self.assertNotExists(test_file('twopart_side.o'))
 
+  def test_tsearch(self):
+    self.run_process([EMCC, test_file('tsearch.c'),
+                      ' -s', 'EXIT_RUNTIME=1'])
+    output = self.run_js('a.out.js')
```
Comment (sbc100) [inline tests/test_other.py:422]: Add new line between emthods.

Comment (sbc100) [inline tests/test_other.py:422]: Replace these twolines with `self.run_other_teast('tsearch.c')`.. then move `tsearch.c` to `tests/other` and create tests/other/test_tseartch.out` alongside it containing some/all of the expected output lines.

I don't think you will need `EXIT_RUNTIME`.

**Review by sbc100 — COMMENTED** [review]
great!

lgtm % comments.

**Comment by msabwat** [comment]
@sbc100 thank you for the review! I pushed --forced the changes into the same branch.

**Review by sbc100 — APPROVED** [review]

File: tests/other/test_tsearch.c [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+		tab[i] = *ptr;
+        val = tsearch((void *) ptr, &root, compare);
+        if (val == NULL)
+            return (1);
+        else if ((*(int **) val) != ptr)
+            free(ptr);
+    }
+    twalk(root, action);
+    tdestroy(root, free);
+    return (0);
```
Comment (sbc100) [inline tests/other/test_tsearch.c:63]: No braces around return value.

File: tests/other/test_tsearch.c [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    if (*(int *) pa < *(int *) pb)
+        return -1;
+    if (*(int *) pa > *(int *) pb)
+        return 1;
+    return 0;
+}
+
+static void action(const void *nodep, VISIT which, int depth) {
+    int *datap;
+
```
Comment (sbc100) [inline tests/other/test_tsearch.c:25]: Just one empty line here.

File: tests/other/test_tsearch.c [hunk 4]
```diff
@@ -0,0 +1,64 @@
+#define _GNU_SOURCE
+
+#include <search.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <time.h>
+#include <assert.h>
+
+static void *root = NULL;
```
Comment (sbc100) [inline tests/other/test_tsearch.c:9]: This can be local in `main` maybe?

**Review by sbc100 — APPROVED** [review]

File: tests/other/test_tsearch.c (hunk 4 above)
Comment (msabwat) [reply tests/other/test_tsearch.c]: yes, done. I push --forced with the new changes, thank you for the review, and sorry for the mistakes

File: tests/other/test_tsearch.c [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        assert(*datap == tab[cnt++]);
+        printf("%6d\n", *datap);
+        break;
+    }
+}
+
+int main(void) {
+    int  *ptr = NULL;
+    void *val = NULL;
+	void *root = NULL;
```
Comment (sbc100) [inline tests/other/test_tsearch.c:45]: Still seeing indentation issues here and lines 49-52.

Comment (msabwat) [reply tests/other/test_tsearch.c]: fixed

File: tests/other/test_tsearch.c [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+int main(void) {
+    int  *ptr = NULL;
+    void *val = NULL;
+    void *root = NULL;
+
+    for (int i = 0; i < 12; i++) {
+        ptr = malloc(sizeof(int));
+        if (!ptr)
+            return 0;
```
Comment (sbc100) [inline tests/other/test_tsearch.c:50]: At least maybe make this `assert(ptr)` we don't want to return 0 (success) when malloc fails.

I'd still prefer if you can completely remove the malloc/free calls and use a fixed array of 12 element insteam.

File: tests/other/test_tsearch.c [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+        *ptr = i & 0xff;
+        tab[i] = *ptr;
+        val = tsearch((void *) ptr, &root, compare);
+        if (val == NULL)
+            return 1;
```
Comment (sbc100) [inline tests/other/test_tsearch.c:55]: How about `assert(val)` instead?

**Review by sbc100 — APPROVED** [review]

File: tests/other/test_tsearch.c (hunk 7 above)
Comment (msabwat) [reply tests/other/test_tsearch.c:55]: better, with malloc gone too

**Comment by sbc100** [comment]
Great!  Thanks for all the revisions.

**Comment by sbc100** [comment]
I can't see why the circleci tests are not running here.  Can you try looking at one of these circleci logs: https://app.circleci.com/jobs/github/emscripten-core/emscripten/431017?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link.   I think it might be that you need to login to circleci before they will run correctly.

**Comment by msabwat** [comment]
> I can't see why the circleci tests are not running here. Can you try looking at one of these circleci logs: https://app.circleci.com/jobs/github/emscripten-core/emscripten/431017?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link. I think it might be that you need to login to circleci before they will run correctly.

done, I hope rebasing will relaunch it, I can't do it from the dashboard

**Comment by sbc100** [comment]
Strange .. circleci still isn't running.

If this problem persists I can maybe land this for you .. although this issue seems very odd.

**Comment by msabwat** [comment]
in progress, seems it's working

**Pull request merged**